### PR TITLE
Prioritize BFD alive status over round trip status

### LINKED ISSUE #2353: BFD: unexpected behavior when half of BFD session previously existed on switch (outside Kilda)
1. Manually (or directly via Floodlight) install bfd session on a switch
   * do not install the peer session. Only one 'part' should exist
2. Request installation of BFD session for an ISL via NB API `PATCH /api/v1/links/enable-bfd`
**Expected:** System does not register BFD session on either end of ISL because there is an existing session
**Actual:** System fails to setup session on busy switch, but does set up a peer session on the other end of ISL and registers it in the DB. It leads to the situation when ISL reacts both to BFD events and to old timeout mechanism. Breaking such ISL leads to the following sequence of events: break ISL -> isl marked as FAILED instantly -> isl is Discovered shortly afterwards -> isl fails again due to discovery timeout

Note: This can appear in live prod environment if previously existed BFD sessions are not removed properly on _both_ ends of ISL

### LINKED ISSUE #3430: failed ISL is not rediscovered after disabling BFD
**Steps to reproduce:**
* find an a-switch ISL with bfd enabled;
* Interrupt ISL connection by breaking rule on a-switch;
* make sure ISL is failed;
* disable bfd;
* restore rule on a-switch;

**Actual result:**
ISL is not rediscovered

**Expected result:**
ISL should be rediscovered

**Workaround**
enable bfd

### PR DESCRIPTION
**Pull request by surabujin**
Route BFD events up to ISL handler (till now they reached only uni-isl
handler) so it can to distinguish BFD events from POLL events and handle
them in another way.

Rework ISL-handler to make prioritization of discovery/alive event
sources.

Along with the main goal this change also fixes some existing issues:
close #2353
close #3430

### CODE REVIEW

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/TickClock.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 import lombok.Getter;
 
+// TODO - del
```
Comment (rozdy) [inline src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/TickClock.java:20]: Why not to delete this class right now?

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/IslEndpointStatus.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import lombok.Data;
 import lombok.Getter;
 
+// TODO - del?
```
Comment (rozdy) [inline src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/IslEndpointStatus.java:23]: Why not to delete this class right now?

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/isl/IslHandler.java [hunk 3]
```diff
@@ -160,12 +161,14 @@ public void islDefaultRulesDelete(Endpoint source, Endpoint destination) {
         emit(STREAM_SPEAKER_RULES_ID, getCurrentTuple(), makeIslRulesDeleteTuple(source, destination));
     }
 
+    // FIXME(surabujin) - storm handler design violation
```
Comment (rozdy) [inline src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/isl/IslHandler.java:164]: Didn't get it. Add some details please.

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/isl/DiscoveryBfdMonitor.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    public Optional<IslStatus> evaluateStatus() {
+        boolean isEnabled = true;
+        boolean isUp = false;
+
+        for (Iterator<IslEndpointBfdStatus> it = discoveryData.stream().iterator(); it.hasNext(); ) {
+            IslEndpointBfdStatus entry = it.next();
+
+            isEnabled &= entry.isEnabled();  // only if both endpoint are BFD capable we can use BFD statuses
+            isUp |= entry.getStatus() == IslStatus.ACTIVE;
+        }
```
[the remark this answers is not in the document]
Comment (surabujin) [reply src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/isl/DiscoveryBfdMonitor.java:49]: It is not simplification - you travel 2 times for each entry i.e. use 2 consecutive loops.

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/isl/IslFsm.java [hunk 5]
```diff
@@ -126,278 +125,229 @@ public IslFsm(Clock clock, PersistenceManager persistenceManager, IslReportFsm r
...
 
-    public void upHandlePollEvent(IslFsmState from, IslFsmState to, IslFsmEvent event, IslFsmContext context) {
-        updateEndpointStatusByEvent(event, context);
-        saveStatusTransaction();
-
-        if (getAggregatedStatus() != IslEndpointStatus.Status.UP) {
-            fire(IslFsmEvent._BECOME_DOWN, context);
+    public void setUpResourcesTimeout(IslFsmState from, IslFsmState to, IslFsmEvent event, IslFsmContext context) {
+        if (--islRulesAttempts >= 0) {
+            log.info("Retrying ISL resoruces setup for {} (attempts lefs:{})", reference, islRulesAttempts);
```
[the remark this answers is not in the document]
Comment (surabujin) [reply src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/isl/IslFsm.java:211]: nice catch

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/TickClock.java (hunk 1 above)
Comment (surabujin) [reply src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/TickClock.java:20]: It is not related to the current change. This is my own reminder, it should not appear in final commit.

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/IslEndpointStatus.java (hunk 2 above)
Comment (surabujin) [reply src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/model/IslEndpointStatus.java:23]: same here

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/isl/IslHandler.java (hunk 3 above)
Comment (surabujin) [reply src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/isl/IslHandler.java:164]: Original design supposes for storm bolt to have `makeStreamNameTuple` methods for each stream. Here we define 2 such methods for 1 stream.

**Pull request merged**